### PR TITLE
Use numpy docstring style everywhere

### LINKED ISSUE #27: Convert docstrings to numpydoc style
... and check the docstring type with flake8. This could be done automatically (up to some fixes :crossed_fingers:) using [pyment](https://github.com/dadadel/pyment).

### PR DESCRIPTION
**Pull request by mlondschien**
Closes #27. I really hope that I did not break any untested code.

Checklist
* [x] Added a `docs/changes.rst` entry

Still undocumented:
 - `minimalkv.decorator.KeyTransformingDecorator` and methods
 - `minimalkv._urls.extract_params`
 - `minimalkv._store_decoration.decorate_store`
 - `minimalkv._boto._get_s3bucket`
 - `minimalkv._store_creation.create_store`
 - `minimalkv._store_creation._create_store_...`
 - `minimalkv._store_creation._build_azure_url`
 - `minimalkv.crypt.HMACDecorator` methods
 - most of `minimalkv.net`.

See occurences of `noqa D`.

### CODE REVIEW

File: minimalkv/_boto.py [hunk 1]
```diff
@@ -5,7 +5,8 @@ def _get_s3bucket(
     secret_key,
     force_bucket_suffix=True,
     create_if_missing=True,
-):
+):  # noqa D
```
Comment (xhochy) [inline minimalkv/_boto.py:8]: This is a private function, so there is no need for a docstring.

File: minimalkv/_hstores.py [hunk 2]
```diff
@@ -22,7 +22,20 @@ class HAzureBlockBlobStore(ExtendedKeyspaceMixin, AzureBlockBlobStore):
...
+        ----------
+        key : str
+            Key of data.
+
+        Returns
+        -------
+        size : int
+            Size of value at key in bytes.
+        """
+        # TODO: Verify docstring.
```
Comment (xhochy) [inline minimalkv/_hstores.py:38]: Docstring is fine 👍

File: minimalkv/crypt.py [hunk 3]
```diff
@@ -109,7 +108,7 @@ def __new_hmac(self, key, msg=None):
 
         return hm
 
-    def get(self, key):
+    def get(self, key):  # noqa D
```
Comment (xhochy) [inline minimalkv/crypt.py:111]: In these cases, we should just inherit the docstring from `KeyValueStore`.

File: minimalkv/decorator.py [hunk 4]
```diff
@@ -9,12 +11,23 @@ class StoreDecorator:
...
+    store: KeyValueStore
+        Store.
     """
 
-    def __init__(self, store):
+    def __init__(self, store: KeyValueStore):
         self._dstore = store
 
     def __getattr__(self, attr):
+        # TODO: Why not use getattr
```
Comment (xhochy) [inline minimalkv/decorator.py:30]: Using `getattr` here would lead to a cycle as `getattr` calls `__getattr__`

**Review by xhochy — COMMENTED** [review]
General remarks:

* There should be no need to document private methods
* We should inherit the doc string from a central place. I know that some OSS project has an `@inherit_docs` decorator for that.

**Comment by mlondschien** [comment]
> There should be no need to document private methods

Fair enough, we can get rid of the `# noqa D`. In general I like adding docstrings even to private methods for the sake of documentation / readability of the code.

> We should inherit the doc string from a central place. I know that some OSS project has an @inherit_docs decorator for that.

How does this work? I'd rather not write some complex docstring parsing code to get this done.

**Comment by xhochy** [comment]
> > There should be no need to document private methods
> 
> Fair enough, we can get rid of the `# noqa D`. In general I like adding docstrings even to private methods for the sake of documentation / readability of the code.
> 
> > We should inherit the doc string from a central place. I know that some OSS project has an @inherit_docs decorator for that.
> 
> How does this work? I'd rather not write some complex docstring parsing code to get this done.

I was thinking of https://github.com/pandas-dev/pandas/blob/244ae516c66659f08fe514f2de2703d3a740cbd7/pandas/util/_decorators.py#L344 where you add `@doc(Parent.method_name)` to inherit the docstring. Maybe we can copy a stripped-down version to this repository. Otherwise, for complicated docstring decorators, you can also reach out to @fjetter :P

**Review by xhochy — APPROVED** [review]

**Pull request merged**